### PR TITLE
statitics: send original string value for sampling data

### PR DESCRIPTION
**Pull request by winoros**
Signed-off-by: Yiding Cui <winoros@gmail.com>

### What problem does this PR solve?

Compabiliable change with https://github.com/pingcap/tidb/pull/25311

Problem Summary:

TiDB still needs the original data instead of the collation key.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

This pr sends the original string value instead of its collation key to TiDB.
This also saves the encoding CPU costs.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Integration test

### Release Note
```
None
```

### CODE REVIEW

File: src/coprocessor/statistics/analyze.rs [hunk 1]
```diff
@@ -461,19 +470,33 @@ impl RowSampleCollector {
...
             }
-            self.fm_sketches[i].insert(&columns_val[i]);
+            if columns_info[i].as_accessor().is_string_like() {
+                self.fm_sketches[i].insert(&collation_keys_val[i]);
+            } else {
+                self.fm_sketches[i].insert(&columns_val[i]);
+            }
             self.total_sizes[i] += columns_val[i].len() as i64;
         }
         self.sampling(columns_val);
```
Comment (skyzh) [inline src/coprocessor/statistics/analyze.rs:502]: So we are sampling on `columns_val` and generating other statistics on `collation_keys_val`?

**Review by skyzh — COMMENTED** [review]
I doubt whether it is necessary to use two separate vectors for `column_vals` and `collation_key_vals`. As they are both of type `Vec<Vec<u8>>`, can we combine them into one?

By the way, please fix clippy errors. See Jenkins for details.

File: src/coprocessor/statistics/analyze.rs (hunk 1 above)
Comment (time-and-fate) [reply src/coprocessor/statistics/analyze.rs:502]: Yes, now we always sample on the original values, calculate the total size using the original values, and build FM Sketch using the collate key (if it has).

Comment (skyzh) [reply src/coprocessor/statistics/analyze.rs:502]: Looks good. Please re-request my review after lint passes.

**Review by skyzh — APPROVED** [review]

**Review by lzmhhh123 — APPROVED** [review]

**Pull request merged**